Editor/GlobalColliderPass.cs

```diff
@@ -0,0 +1,260 @@
+﻿#region
+
+using System;
+using System.Collections.Generic;
+using nadena.dev.modular_avatar.editor.ErrorReporting;
+using nadena.dev.ndmf;
+
+#if MA_VRCSDK3_AVATARS
+using nadena.dev.ndmf.vrchat;
+using static VRC.SDK3.Avatars.Components.VRCAvatarDescriptor;
+#endif
+
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+#endregion
+
+namespace nadena.dev.modular_avatar.core.editor
+{
+	internal class GlobalColliderPass : Pass<GlobalColliderPass>
+	{
+		protected override void Execute(ndmf.BuildContext ctx)
+		{
+			var remapColliders = ctx.AvatarRootTransform.GetComponentsInChildren<ModularAvatarGlobalCollider>(true);
+			if (remapColliders.Length == 0) return;
+
+			Array.Sort(remapColliders, (a, b) =>
+			{
+				if (a.manualRemap && !b.manualRemap) return -1;
+				if (!a.manualRemap && b.manualRemap) return 1;
+				return 0;
+			});
+
+#if MA_VRCSDK3_AVATARS
+			var indexFingerWarns = new List<GameObject>();
+			var skippedColliders = new List<GameObject>();
+
+			var usedColliders = new List<GlobalCollider>();
+			var colliderPriority = new[]
+			{
+				GlobalCollider.FingerRingLeft,
+				GlobalCollider.FingerRingRight,
+				GlobalCollider.FingerMiddleLeft,
+				GlobalCollider.FingerMiddleRight,
+				GlobalCollider.FingerLittleLeft,
+				GlobalCollider.FingerLittleRight,
+				GlobalCollider.FingerIndexLeft,
+				GlobalCollider.FingerIndexRight
+			};
+
+			var ColliderIndex = 0;
+			foreach (var my in remapColliders)
+			{
+				ColliderIndex++;
+				GlobalCollider targetCollider;
+
+				//If none, use gameobject component is on.
+				var remapTargetObj = my.remapTargetObject ?? my.gameObject;
+
+				var newColliderConfig = new ColliderConfig {
+					state = ColliderConfig.State.Custom,
+					isMirrored = false,
+					transform = remapTargetObj.transform,
+					radius = my.radius,
+					height = my.height,
+					position = my.position,
+					rotation = my.rotation,
+				};
+
+				if (my.manualRemap)
+				{
+					//This ended up being overly instrusive since it's purposefully used in some of my workflows. I don't think it's needed.
+					/*if (usedColliders.Contains(my.sourceCollider))
+					{
+						//Collider was already used, so it will be overriten by this remap.
+						//BuildReport.Log(ErrorSeverity.Information, "validation.global_collider.manual_collider_overwrite", my.gameObject);
+					}*/
```

Comment:
This should probably be a warning. It would be hard to debug otherwise, after all (and it should cite both components, not just one)